ginivision/src/main/java/net/gini/android/vision/camera/FrameView.java

```diff
@@ -0,0 +1,139 @@
+package net.gini.android.vision.camera;
+
+
+import android.content.Context;
+import android.graphics.Canvas;
+import android.graphics.Color;
+import android.graphics.Paint;
+import android.support.annotation.Nullable;
+import android.util.AttributeSet;
+import android.view.View;
+
+public class FrameView extends View {
+
+    private static final int LINE_LENGTH = 70;
+    private static final int LINE_WIDTH = 1;
+
+    private int mHeight;
+    private float mLineLength;
+    private Paint mPaintLine;
+    private int mWallOffsetSide;
+    private int mWallOffsetTop;
+    private int mWidth;
+    private boolean mPortrait;
+
+    public FrameView(final Context context) {
+        this(context, null);
+    }
+
+    public FrameView(final Context context, @Nullable final AttributeSet attrs) {
+        super(context, attrs);
+
+        mPaintLine = new Paint();
+        mPaintLine.setColor(Color.WHITE);
+        mPaintLine.setStyle(Paint.Style.STROKE);
+
+        setLineWidth(LINE_WIDTH);
+        setLineLength(LINE_LENGTH);
+    }
+
+    @Override
+    protected void onDraw(final Canvas canvas) {
+        super.onDraw(canvas);
+        drawShadowRectangles(canvas);
+        drawUpperLeftLines(canvas);
+        drawLowerLeftLines(canvas);
+        drawUpperRightLines(canvas);
+        drawLowerRightLines(canvas);
+        drawBackgroundForButtons(canvas);
+    }
+
+    private void drawBackgroundForButtons(final Canvas canvas) {
+        if(mPortrait) {
+            final Paint paintRectangle = new Paint();
+            paintRectangle.setStyle(Paint.Style.FILL);
+            paintRectangle.setColor(Color.BLACK);
+            paintRectangle.setAlpha(75);
+
+            canvas.drawRect(mWallOffsetSide, mHeight-mWallOffsetTop -mLineLength, mWidth - mWallOffsetSide, mHeight - mWallOffsetTop, paintRectangle);
+        }
+    }
+
+    @Override
+    protected void onMeasure(final int widthMeasureSpec, final int heightMeasureSpec) {
+        mWidth = MeasureSpec.getSize(widthMeasureSpec);
+        mHeight = MeasureSpec.getSize(heightMeasureSpec);
+
+        mPortrait = mWidth < mHeight;
+
+        mWallOffsetTop = dpToPx(20);
```

Comment:
In landscape the the corners should be shown also in landscape. I switched setting the offsets around for landscape and it worked:
```
        if (mPortrait) {
            mWallOffsetTop = dpToPx(20);
            int documentHeight = mHeight - mWallOffsetTop;
            int documentWidth = (int) (documentHeight / Math.sqrt(2)); //din a4
            mWallOffsetSide = (mWidth - documentWidth) / 2;
        } else {
            mWallOffsetSide = dpToPx(20);
            int documentWidth = mWidth - mWallOffsetSide;
            int documentHeight = (int) (documentWidth / Math.sqrt(2)); //din a4
            mWallOffsetTop = (mHeight - documentHeight) / 2;
        }
```